### PR TITLE
Update icu for 67.1

### PR DESCRIPTION
**Pull request by foopoiuyt**
Update ICU port to 67.1 pulling in post-67.1 patch for separating out source files and moving the updated cmake bits into a patch. Former is in upstream icu, latter still needs some work for other uses (for example, being able to run the tests and samples) but should be sufficient for our current needs.

### CODE REVIEW

File: ports/icu/portfile.cmake [hunk 1]
```diff
@@ -103,5 +108,20 @@ endif ()
...
+file(COPY
+    DESTINATION ${CURRENT_PACKAGES_DIR}/share/icu
+    PATTERN ${CURRENT_PACKAGES_DIR}/lib/cmake/*.cmake
+)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/cmake)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/cmake)
+
 file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/icu RENAME copyright)
-file(WRITE ${CURRENT_PACKAGES_DIR}/share/icu/version "${VERSION_MAJOR}.${VERSION_MINOR}.0")
+file(WRITE ${CURRENT_PACKAGES_DIR}/share/icu/version "${ICU_VERSION_MAJOR}.${ICU_VERSION_MINOR}.0")
```
Comment (donny-dont) [inline ports/icu/portfile.cmake:127]: All the `FOO_VERSION` bits were renamed to just be `VERSION` so this line shouldn't change.

File: ports/icu/portfile.cmake [hunk 2]
```diff
@@ -69,32 +68,38 @@ vcpkg_copy_pdbs()
...
+        icupkg)
 
     set(TOOL_EXTENSION .exe)
 
     file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools/icu)
 
     foreach (tool ${TOOLS})
         # Remove debug versions
-        file(REMOVE ${CURRENT_PACKAGES_DIR}/debug/bin/${tool}${TOOL_EXTENSION})
+        file(REMOVE ${CURRENT_PACKAGES_DIR}/debug/bin64/${tool}${TOOL_EXTENSION})
```
Comment (donny-dont) [inline ports/icu/portfile.cmake:94]: This part will fail when building anything outside of x64.

Comment (foopoiuyt) [reply ports/icu/portfile.cmake]: Updated to not suffix the path and change back to just bin here.

**Comment by foopoiuyt** [comment]
Both of the above should be resolved now.

**Pull request merged**